src/sentry/eventstore/reprocessing/redis.py

```diff
@@ -0,0 +1,166 @@
+import uuid
+from collections.abc import Sequence
+from datetime import datetime
+from typing import Any
+
+import redis
+from django.conf import settings
+
+from sentry.utils import json
+from sentry.utils.dates import to_datetime, to_timestamp
+from sentry.utils.redis import redis_clusters
+
+from .base import ReprocessingStore
+
+
+def _get_sync_counter_key(group_id):
+    return f"re2:count:{group_id}"
+
+
+def _get_info_reprocessed_key(group_id):
+    return f"re2:info:{group_id}"
+
+
+def _get_old_primary_hash_subset_key(project_id: int, group_id: int, primary_hash: str):
+    return f"re2:tombstones:{{{project_id}:{group_id}:{primary_hash}}}"
+
+
+class RedisReprocessingStore(ReprocessingStore):
+    def __init__(self, **options) -> None:
+        self.redis = redis_clusters.get(options.pop("cluster", "default"))
```

Comment:
Technically `self.redis` can be `RedisCluster` and `StrictRedis`, once https://github.com/getsentry/sentry/pull/66146 lands there might be a type issue